cjwkernel/types.py

```diff
@@ -582,6 +582,67 @@ def _i18n_argument_to_thrift(value: Union[str, int, float]) -> ttypes.I18nArgume
         raise RuntimeError("Unhandled value for I18nArgument: %r" % value)
 
 
+@dataclass(frozen=True)
+class I18nMessageSource:
+    @classmethod
+    def from_thrift(
+        cls, value: ttypes.I18nMessageSource
+    ) -> Optional[I18nMessageSource]:
+        if value.module_id is not None:
+            return I18nMessageSource.Module(value.module_id)
+        elif value.library is not None:
+            return I18nMessageSource.Library(value.library)
+        else:
+            return None
+
+    @abstractmethod
```

Comment:
This is not an abstract method. Nix `@abstractmethod` and add an `# override` comment instead.